### PR TITLE
Extended tool for subdirectories

### PR DESCRIPTION
**Pull request by shashwat1998**
xml files stored in directories and sub directories can now be converted.

### CODE REVIEW

File: objectmapper.py [hunk 1]
```diff
@@ -25,14 +25,16 @@ def __init__(self):
...
+        ann.filename = xml_file_path
+        return ann
 
-    def bind_files(self, xml_file_paths):
+    def bind_files(self, xml_file_paths, xml_dir):
         result = []
         for xml_file_path in xml_file_paths:
             try:
-                result.append(self.bind(xml_file_path=xml_file_path))
+                result.append(self.bind(xml_file_path=xml_file_path,xml_dir=xml_dir))
```
Comment (isabek) [inline objectmapper.py]: Please, format the code

File: reader.py [hunk 2]
```diff
@@ -7,9 +7,12 @@ def __init__(self, xml_dir):
...
     def get_xml_files(self):
         xml_filenames = []
-        for xml_filename in os.listdir(self.xml_dir):
-            if xml_filename.endswith(".xml"):
-                xml_filenames.append(os.path.join(self.xml_dir, xml_filename))
+        for root, subdirs, files in os.walk(self.xml_dir):
+            for filename in files:
+                if filename.endswith(".xml"):
+                    file_path = os.path.join(root, filename)
+                    file_path = os.path.relpath(file_path, start = self.xml_dir)
```
Comment (isabek) [inline reader.py]: Please, format the code.

File: transformer.py [hunk 3]
```diff
@@ -14,12 +14,15 @@ def transform(self):
         xml_files = reader.get_xml_files()
         classes = reader.get_classes()
         object_mapper = ObjectMapper()
-        annotations = object_mapper.bind_files(xml_files)
+        annotations = object_mapper.bind_files(xml_files,xml_dir=self.xml_dir)
```
Comment (isabek) [inline transformer.py]: Please, format the code.

**Comment by shashwat1998** [comment]
I'm sorry didn't understand what exactly do you mean. Do you want better comments and readable code?

**Comment by isabek** [comment]
@shashwat1998 I use the default Code Style scheme of Pycharm. Please, use it for formatting your code. After comma should be blank space. Also, there shouldn't be spaces around `=` symbol.

**Comment by isabek** [comment]
@shashwat1998 thanks for your contribution.

**Pull request merged**